js/libs/ui-shared/src/user-profile/LocaleSelector.tsx

```diff
@@ -12,17 +12,21 @@ const localeToDisplayName = (locale: string) => {
 
 type LocaleSelectorProps = Omit<UserProfileFieldProps, "inputType"> & {
   supportedLocales: string[];
+  currentLocale: string;
 };
 
 export const LocaleSelector = ({
   t,
   form,
   supportedLocales,
+  currentLocale,
 }: LocaleSelectorProps) => {
-  const locales = supportedLocales.map((locale) => ({
-    key: locale,
-    value: localeToDisplayName(locale) || "",
-  }));
+  const locales = supportedLocales
+    .map((locale) => ({
+      key: locale,
+      value: t(`locale_${locale}`, localeToDisplayName(locale) ?? locale),
+    }))
+    .sort((a, b) => a.value.localeCompare(b.value, currentLocale));
```

Comment:
I am thinking, since this is likely an expensive operation to do on each render, we might want to memoize this.
```suggestion
  const locales = useMemo(
    () =>
      supportedLocales
        .map((locale) => ({
          key: locale,
          value: t(`locale_${locale}`, localeToDisplayName(locale) ?? locale),
        }))
        .sort((a, b) => a.value.localeCompare(b.value, currentLocale)),
    [supportedLocales, currentLocale, t],
  );

```